src/Microsoft.JSInterop/DotNetObject.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.JSInterop
+{
+    /// <summary>
+    /// Wraps a JS interop argument, indicating that the value should not be serialized as JSON
+    /// but instead should be passed as a reference.
+    ///
+    /// To avoid leaking memory, the reference must later be disposed by JS code.
+    /// </summary>
+    public struct DotNetObject
```

Comment:
Is there a benefit to making this a struct? It looks like it's always going to be boxed.

Reply:
It will always be boxed when it's being passed to `TrySerializeKnownTypes`, but doesn't have to be boxed at other times.

Originally I was going to make it a reference type because then we could preserve the same box, but since it turns out we're not storing `DotNetObject` anywhere, it was not clear that there'd be any advantage.

Overall I'm not seeing any case where it would make any difference either way, so I erred on the side of not forcing extra heap objects if there are any cases where developers want to make, say, an array or list of them.

If you feel there's any reason to prefer having it as a reference type I'm happy to change it.